File: Programmers/[60057] 문자열 압축.py
def solution(s):
    result = [len(s)]

    for i in range(1, len(s)//2 + 1):
        temp = ""
        cnt = 1
        sub_s = s[0:i]

        for j in range(i, len(s), i):
            if sub_s == s[j:j+i]:
                cnt += 1
            else:
                if cnt == 1:
                    temp += sub_s
                else:
                    temp += str(cnt) + sub_s
                sub_s = s[j:j+i]
                cnt = 1

        if cnt == 1:
            temp += sub_s
        else:
            temp += str(cnt) + sub_s
        result.append(len(temp))

    return min(result)

File: Programmers/[60057] 문자열 압축.java
package pg60057;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

class Solution {
    public int solution(String s) {
        int sLength = s.length();
        List<Integer> result = new ArrayList<>();
        result.add(sLength);

        for (int i = 1; i <= sLength / 2; i++) {
            StringBuilder regex = new StringBuilder("(");
            regex.append(".".repeat(Math.max(0, i)));
            regex.append(")");

            Pattern pattern = Pattern.compile(String.valueOf(regex));
            Matcher match = pattern.matcher(s);
            List<String> sList = new ArrayList<>();
            while (match.find()) {
                sList.add(match.group());
            }
            if (sLength % i != 0) {
                sList.add(s.substring((sLength / i) * i, sLength));
            }

            StringBuilder compressedStr = new StringBuilder();
            for (int j = 0; j < sList.size(); j++) {
                int sameWordCnt = 1;
                for (int k = j + 1; k < sList.size(); k++) {
                    if (!sList.get(j).equals(sList.get(k))) {
                        break;
                    }
                    sameWordCnt++;
                }

                if (sameWordCnt > 1) {
                    compressedStr.append(sameWordCnt);
                }
                compressedStr.append(sList.get(j));
                j += sameWordCnt - 1;
            }
            result.add(compressedStr.length());
        }

        return Collections.min(result);
    }
}